pyanaconda/core/configuration/anaconda.py

```diff
@@ -71,6 +72,159 @@ def kickstart_modules(self):
+    """The Installation System section."""
+
+    @property
+    def _type(self):
+        """Type of the installation system.
+
+        FIXME: This is a temporary solution.
+        """
+        return self._get_option("type", SystemType)
+
+    @property
+    def _is_boot_iso(self):
+        """Are we running in the boot.iso?"""
+        return self._type is SystemType.BOOT_ISO
+
+    @property
+    def _is_live_os(self):
+        """Are we running in the live OS?"""
+        return self._type is SystemType.LIVE_OS
+
+    @property
+    def _is_unknown(self):
+        """Are we running in the unknown OS?"""
+        return self._type is SystemType.UNKNOWN
+
+    @property
+    def can_reboot(self):
+        """Can we reboot the system?"""
+        return self._is_boot_iso
+
+    @property
+    def can_switch_tty(self):
+        """Can we change the foreground virtual terminal?"""
+        return self._is_boot_iso
+
+    @property
+    def can_audit(self):
+        """Can we run the audit daemon?"""
+        return self._is_boot_iso
+
+    @property
+    def can_adjust_time(self):
+        """Can we change the time?"""
+        return self._is_boot_iso
+
+    @property
+    def can_adjust_time_live(self):
+        """Can we change the time?
+
+        FIXME: Conflict with can_adjust_time.
+        """
+        return self._is_boot_iso or self._is_live_os
+
+    @property
+    def can_synchronize_time(self):
+        """Can we run the NTP daemon?"""
+        return self._is_boot_iso
+
+    @property
+    def can_localize(self):
+        """Can we change the localization?"""
+        return self._is_boot_iso
+
+    @property
+    def can_localize_live(self):
+        """Can we change the localization?
+
+        FIXME: Conflict with can_localize.
+        """
+        return self._is_boot_iso or self._is_live_os
+
+    @property
+    def can_configure_syslog(self):
+        """Can we modify syslog?
+
+        FIXME: This rule is weird.
+        """
+        return self._is_boot_iso
+
+    @property
+    def can_modify_nvram(self):
+        """Can we modify firmware NVRAM variables?
+
+        FIXME: Isn't this target specific?
+        """
+        return self._is_boot_iso or self._is_live_os
+
+    @property
+    def can_disable_swap(self):
+        """Can we call swapoff?
+
+        FIXME: This should be safe to do on the boot.iso.
+        FIXME: This rule seems to be too specific.
+        """
+        return self._is_live_os
+
+    @property
+    def can_copy_resolve_conf(self):
+        """Can we copy /etc/resolv.conf to the target system?
+
+        FIXME: Isn't this rule too specific?
+        """
+        return self._is_boot_iso
+
+    @property
+    def can_change_hostname(self):
+        """Can we change the hostname?"""
+        return self._is_boot_iso
+
+    @property
+    def can_change_hostname_live(self):
+        """Can we change the hostname?
+
+        FIXME: Conflict with can_change_hostname.
+        """
```

Comment:
For the hostname, we need to talk...
I hope we could do with just one check (the one in network module) - can_change_hostname, which would be true also for _is_live_os though.
I think we should allow/be able to (with regard to current hostname):
BOOT_ISO: get and set
LIVE_CD: get - just for storage containers naming (set does not make sense as there is no Network Spoke and setting in write_network_configuration is dubious)
UNKNOWN (ie OS with dirinstall/imageinstall): nothing
INITIAL SETUP: get and set ? (reconfig mode, ARM images configuration)

Reply:
Replaced with `can_change_hostname`, because it wasn't possible to change a hostname on live OS anyway.

Reply:
I have added a new rule `provides_system_bus`, that is used in the network DBus module instead of `can_change_hostname`. The rule `can_change_hostname` is checked only before changing the hostname.